cmd/clusterctl/client/config.go

```diff
@@ -58,6 +58,38 @@ func (c *clusterctlClient) GetProviderComponents(provider string, providerType c
 	return components, nil
 }
 
+type ProcessYAMLOptions struct {
+	// URLSource to be used for reading the template
+	URLSource *URLSourceOptions
+
+	// ListVariablesOnly return the list of variables expected by the template
+	// without executing any further processing.
+	ListVariablesOnly bool
+}
+
+func (c *clusterctlClient) ProcessYAML(options ProcessYAMLOptions) (YamlPrinter, error) {
+
+	// Technically we do not need to connect to the cluster. However, we are
+	// leveraging the template client which exposes GetFromURL() is available
+	// on the cluster client so we create a cluster client with default
+	// configs to access it.
+	cluster, err := c.clusterClientFactory(
+		ClusterClientFactoryInput{
+			// use the default kubeconfig
+			kubeconfig: Kubeconfig{},
+		},
+	)
```

Comment:
Since `GetFromURL` has nothing to do with a cluster, "ideally" we would refactor it out into a standalone template client but that is quite a decent amount of refactoring. I opted to do it this way to reuse as much of the existing logic and not blow this PR out of proportions.